pandas/tests/test_strings.py

```diff
@@ -558,50 +551,56 @@ def test_extract(self):
 
             # single group renames series/index properly
             s_or_idx = klass(['A1', 'A2'])
-            result = s_or_idx.str.extract(r'(?P<uno>A)\d')
-            tm.assert_equal(result.name, 'uno')
-            tm.assert_numpy_array_equal(result, klass(['A', 'A']))
+            result_df = s_or_idx.str.extract(r'(?P<uno>A)\d')
+            result_series = result_df['uno']
+            tm.assert_numpy_array_equal(result_series, klass(['A', 'A']))
+
+    def test_extract_series(self):
+        # extract should give the same result whether or not the
+        # series has a name.
+        for series_name in None, "series_name":
```

Comment:
here is the test that makes sure extract gives the same result for a series with and without a name.